src/routes/movies.py

```
@@ -225,108 +515,210 @@
+            "directors": [director.name for director in movie.directors],
+        }
 
-    except IntegrityError:
+        return MovieDetailSchema.model_validate(movie_dict)
+
+    except IntegrityError as e:
+        await db.rollback()
+        raise HTTPException(
+            status_code=400,
+            detail=f"Database integrity error: {str(e)}"
+        )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)